[material-next][ButtonGroup] Copy ButtonGroup to material next

lhilgert9 opened the pull request:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

**ButtonGroup issue:** https://github.com/mui/material-ui/issues/39686
**Material You umbrella issue:** https://github.com/mui/material-ui/issues/29345

## Changes
- Copy ButtonGroup from `material` v5
- Fix imports

lhilgert9 commented:
I think the errors in the tests are caused by the different ButtonGroupContexts in the Button and the ButtonGroup. I would suggest to delete the .test file for this PR and add it again in a later PR when the new ButtonGroupContext is implemented in the Button and ButtonGroup of `material-next`.

DiegoAndai reviewed (commented):
Hey @lhilgert9, thanks for working on this 🎉 

About the failing tests, I think these imports should be changed, would that fix it?

DiegoAndai commented:
> About the failing tests, I think these exports should be changed, would that fix it?

Answering to myself: it probably breaks other tests because these changes https://github.com/mui/material-ui/pull/38520 need to be ported to `material-next`'s Button. We could port the [Button's changes on that PR](https://github.com/mui/material-ui/pull/38520/files#diff-2e6df3f03bdd7cef475f4362270feda335c131b3acb55ba73b0f33b3cd4ea357) to `material-next`'s Button in this PR.

DiegoAndai commented:
> In the main PR https://github.com/mui/material-ui/pull/39699 I implemented this and fixed the imports in the .test file, but there I have an error somewhere, which I have not yet found.

Let's implement ButtonGroupContext in `material-next`'s Button and update the imports in this PR. Then, we can work together on trying to figure out the error you're facing.

lhilgert9 commented:
> > In the main PR #39699 I implemented this and fixed the imports in the .test file, but there I have an error somewhere, which I have not yet found.
> 
> Let's implement ButtonGroupContext in `material-next`'s Button and update the imports in this PR. Then, we can work together on trying to figure out the error you're facing.

So you want to continue in this PR because I've already completely done this work to the other PR. As I said, it only depends on the error that occurs during the test. Maybe we could also just try to find the error in PR #39699

mj12albert commented:
> So you want to continue in this PR

Yes ~ we'll be able to review and iterate much faster than one big PR!

mj12albert commented:
Replaced by https://github.com/mui/material-ui/pull/39739
